flutter_frontend_server/lib/server.dart

```diff
@@ -169,3 +182,59 @@ Future<int> starter(
   frontend.listenAndCompile(compiler, input ?? stdin, options, completer);
   return completer.future;
 }
+
+/// A [RecursiveVisitor] that replaces [Object.toString] overrides with
+/// `super.toString()`.
+class ToStringVisitor extends RecursiveVisitor<void> {
+  /// The [packageUris] must not be null.
+  ToStringVisitor(this._packageUris) : assert(_packageUris != null);
+
+  /// A set of package URIs to apply this transformer too, e.g. 'dart:ui' and
+  /// 'package:flutter/foundation.dart'.
+  final Set<String> _packageUris;
+
+  @override
+  void visitProcedure(Procedure node) {
+    if (node.name.name == 'toString' && node.enclosingLibrary != null) {
```

Comment:
nit: You can flip the sense of the ifs to lower the nesting depth.

```dart
if (node.name.name != 'toString' || node.enclosingLibrary == null) {
  super.visitProcedure(node);
  return;
}
assert(node.enclosingClass != null);
// Turn 'dart:ui' into 'dart:ui', or
// 'package:flutter/src/semantics_event.dart' into 'package:flutter'.
final String packageUri =
  '${node.enclosingLibrary.importUri.scheme}:'
  '${node.enclosingLibrary.importUri.pathSegments.first}';
if (!_packageUris.contains(packageUri)) {
  super.visitProcedure(node);
  return;
}
node.function.replaceWith(
  FunctionNode(
    ReturnStatement(
      SuperMethodInvocation(
        node.name,
        Arguments(<Expression>[]),
      ),
    ),
  ),
);
super.visitProcedure(node);
```

Reply:
I did this slightly differently so that I don't have to do multiple returns but still dropped some nesting.

This also made me realize it's a mistake to assert enclosingClass is null - I should be checking for it instead, in case someone put a `toString` method at the top of their library.  Added a test for that.